website/docs/cloud-docs/api-docs/run-tasks/run-tasks.mdx

```diff
@@ -429,12 +429,13 @@ This POST endpoint requires a JSON object with the following properties as a req
 
 Properties without a default value are required.
 
-| Key path                            | Type   | Default | Description                                                                         |
-| ----------------------------------- | ------ | ------- | ----------------------------------------------------------------------------------- |
-| `data.type`                         | string |         | Must be `"workspace-tasks"`.                                                        |
-| `data.attributes.enforcement-level` | string |         | The enforcement level of the workspace task. Must be `"advisory"` or `"mandatory"`. |
-| `data.relationships.task.data.id`   | string |         | The ID of the run task.                                                             |
-| `data.relationships.task.data.type` | string |         | Must be `"tasks"`.                                                                  |
+| Key path                            | Type   | Default       | Description                                                                         |
```

Comment:
> First should this be "Associating" a run task with a workspace? On the actual run tasks docs page, we say associate, not attach.

Good catch. I'll get that changed.

> Also, I think at the top of this section right under the Attach to Workspace header, we should repeat the bit about how you can have run tasks execute at different states in the run lifecycle. this is where we put our beta note

After looking at PR as a whole, I'm finding that we're repeating ourselves a lot; Listing what stages are available and explaining what each stage is.  Is it possible we can just cross post a lot more?  I know being verbose and keeping the user "in the flow" of the document is important, but, as you can see, it's quite a maintenance burden.

I do really like your example, but I also know that there are plans to add one (or even two) more Task stages so that sentence is going to get a bit long listing out all of the possible options.

Reply:
How about this- https://github.com/hashicorp/terraform-docs-common/pull/72/commits/72a95a9d6e8d0bc39f026f960de6a5bb1165c613

Reply:
I like that! I know - I'm struggling with that a bit. There is definitely a balance between making sure people know what's going on and maintenance burden.....I just always worry that folks skim quickly to what they need and if we don't provide any context there/there is no way to get that context (no linking or anything) people will be sad and confused. I appreciate you pushing back on me though - I don't want to make this so ridiculous that we say the same thing every few sections, haha.